ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/results/DebugStorageRangeAtResult.java

```diff
@@ -43,17 +49,38 @@ public String getNextKey() {
     return nextKey;
   }
 
+  @JsonGetter(value = "complete")
+  public boolean getComplete() {
+    return nextKey == null;
+  }
+
   @JsonPropertyOrder(value = {"key", "value"})
   public static class StorageEntry {
+    private final String key;
     private final String value;
 
-    public StorageEntry(final UInt256 value) {
-      this.value = value.toHexString();
+    public StorageEntry(final BytesValue key, final UInt256 value) {
+      this.key = key == null ? null : strictShortHex(key.toString());
+      this.value = strictShortHex(value.toHexString());
+    }
+
+    String strictShortHex(final String hexString) {
+      String workString = hexString;
+      if (workString.startsWith(("0x"))) {
+        workString = workString.substring(2);
+      }
+      while (workString.startsWith("00")) {
+        workString = workString.substring(2);
+      }
+      if (workString.length() == 0) {
+        workString = "00";
+      }
+      return "0x" + workString;
```

Comment:
While this code is really easy to understand and grok, a great naive implementation, given a bit more thought, I feel like this could be done a lot more efficiently and with fewer lines of code.